apps/prairielearn/src/sprocs/sync_assessments.sql

```diff
@@ -387,6 +388,17 @@ BEGIN
                             computed_max_auto_points := (assessment_question->>'max_points')::double precision;
                         END IF;
                     END IF;
+
+                    IF (assessment_question->>'question_id')::bigint IS NULL THEN 
+                        -- TODO this should NEVER be hit in production, because it is inserting 
+                        -- fake questions into the question table to make syn succeed
```

Comment:
👀 TODO to be resolved

```suggestion
                        -- TODO this should NEVER be hit in production, because it is inserting 
                        -- fake questions into the question table to make sync succeed
```

Reply:
This TODO was actually something I needed to ask you about, rather than something I needed to do. 

Are you ok with the way that this query is inserting fake values into the Questions table here? 

The idea is that it should never be hit in production, because in production you would never have an assessment_question with a null question_Id like we have here, because in production the config value `checkSharingOnSync` to be true.

Is there a better way we can put guardrails around this to make sure nothing bad happens? or is just leaving this if statement in this query ok?

Reply:
I added an if statement to raise an exception if this code is ever reached when `config.checkSharingOnSync` is true. That should give us a little extra protection at least.